### PR TITLE
#3593 Create the relation between paper cells and developer tools 

### PR DESCRIPTION
**Pull request by konekoya**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3593)
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/DevTool/DevToolDialog.js [hunk 1]
```diff
@@ -161,14 +200,32 @@ const DevToolDialog = () => {
     [data.topicName, currentWorkspace],
   );
 
+  const prevTopicName = usePrevious(data.topicName);
   useEffect(() => {
     if (isEmpty(data.topicName) || isEmpty(currentWorkspace)) return;
+    if (prevTopicName === data.topicName) return;
```
Comment (konekoya) [inline ohara-manager/client/src/components/DevTool/DevToolDialog.js]: Don't fetch the same topic if it's already fetched

File: ohara-manager/client/src/components/DevTool/DevToolDialog.js [hunk 2]
```diff
@@ -114,19 +124,48 @@ const listCache = new CellMeasurerCache({
...
+    // The selected cell could be other connector types do a quick check here
+    if (!selectedCell || !isOpen) return;
+
+    const isTopic = get(selectedCell, 'classType', null) === 'topic';
+
+    if (isTopic) {
+      if (data.type === 'logs') {
+        return setDataDispatch({
+          topicName: selectedCell.name,
+          service: 'broker',
```
Comment (konekoya) [inline ohara-manager/client/src/components/DevTool/DevToolDialog.js]: Under the _logs_ tab, a selected topic should fetch broker logs

File: ohara-manager/client/src/components/DevTool/Header/Header.js [hunk 3]
```diff
@@ -232,55 +162,102 @@ const Header = props => {
...
-          <RefreshIcon className="item" onClick={handleRefresh} />
-          <SearchIcon className="item" onClick={handleSearchClick} />
+
+          <IconButton
+            className="item"
+            disabled={isButtonDisabled(data)}
+            onClick={handleRefresh}
+            size="small"
+          >
+            <Tooltip title="Fetch the data again" enterDelay={1000}>
```
Comment (konekoya) [inline ohara-manager/client/src/components/DevTool/Header/Header.js]: Add some tooltips here to assistant our users

File: ohara-manager/client/src/components/Pipeline/Graph/Graph.js [hunk 4]
```diff
@@ -58,7 +60,7 @@ const Graph = props => {
   let dragStartPosition = useRef(null);
   let currentCell = useRef(null);
 
-  useEffect(() => {
+  useMountEffect(() => {
```
Comment (konekoya) [inline ohara-manager/client/src/components/Pipeline/Graph/Graph.js]: Re-render Graph will lose all the cells added into the Graph, so we're disabling it here

File: ohara-manager/client/src/components/Pipeline/Toolbox/toolboxUtils.js [hunk 5]
```diff
@@ -258,8 +260,8 @@ export const enableDragAndDrop = params => {
           // once the real graph is added
           const params = {
             position: { x: newX, y: newY },
-            isTemporary: true, // A temp graph
-            title: 'newgraph',
+            isTemporary: !isTopic, // Display a temp cell for all connectors but topic
+            title: !isTopic ? 'newgraph' : displayName, // Same here, display a temp name for all connectors except topic
```
Comment (konekoya) [inline ohara-manager/client/src/components/Pipeline/Toolbox/toolboxUtils.js]: Topic doesn't need a temporary cell

File: ohara-manager/client/src/components/Pipeline/Toolbox/toolboxUtils.js [hunk 6]
```diff
@@ -274,10 +276,10 @@ export const enableDragAndDrop = params => {
             },
           };
 
-          const newCell =
-            classType === 'topic' ? TopicGraph(params) : ConnectorGraph(params);
+          if (!isTopic) openAddConnectorDialog(true);
```
Comment (konekoya) [inline ohara-manager/client/src/components/Pipeline/Toolbox/toolboxUtils.js]: And don't open the dialog as Topic is not needed

File: ohara-manager/client/src/components/Pipeline/Graph/Graph.js [hunk 7]
```diff
@@ -40,6 +40,8 @@ const Graph = props => {
   } = useZoom();
 
   const { setCenter, isCentered, setIsCentered } = useCenter();
+  const { setSelectedCell } = usePipelineActions();
+  const { selectedCell } = usePipelineState();
```
Comment (konekoya) [inline ohara-manager/client/src/components/Pipeline/Graph/Graph.js]: Add a new state which tracks selected cell and shares with devTool component

File: ohara-manager/client/src/components/DevTool/Header/Header.js [hunk 8]
```diff
@@ -232,55 +162,102 @@ const Header = props => {
...
-          <DialogSelect
+          <DevToolSelect
             index={tabName.topic}
             currentTab={tabIndex}
             value={data.topicName}
-            onChange={handleSelectTopic}
+            onChange={event => handleSelect(event, 'topicName')}
             list={topics.map(topic => topic.settings.name)}
             setAnchor={handleSelectAnchor}
             anchor={selectAnchor}
```
Comment (saivirtue) [inline ohara-manager/client/src/components/DevTool/Header/Header.js]: When clicking refresh icon, I think we should disable `select` also.

File: ohara-manager/client/src/components/Layout/AppBar.js [hunk 9]
```diff
@@ -123,7 +127,12 @@ const AppBar = () => {
             onClick={openListWorkspacesDialog}
           />
           <AssignmentIcon className="item" />
-          <DeveloperModeIcon className="item" onClick={openDevToolDialog} />
+          <DeveloperModeIcon
+            className="item"
+            onClick={
+              isDevToolDialogOpen ? closeDevToolDialog : openDevToolDialog
```
Comment (konekoya) [inline ohara-manager/client/src/components/Layout/AppBar.js]: Toggling devTool dialog /cc @eechih

File: ohara-manager/client/src/components/DevTool/Header/Header.js (hunk 8 above)
Comment (konekoya) [reply ohara-manager/client/src/components/DevTool/Header/Header.js]: Good catch, I will update the PR :)

Comment (konekoya) [reply ohara-manager/client/src/components/DevTool/Header/Header.js]: I've done some fixes in this PR and @wu87988622 is waiting this PR. So I will merge this PR first, the issue you pointed out will be addressed in #3616

**Pull request merged**